app/basicComponents/CopyButton.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import styled from 'styled-components';
+
+type Props = {
+  secondary?: boolean;
+  onClickCopy: React.MouseEventHandler<HTMLImageElement>;
+  children?: JSX.Element;
+  isCopied?: boolean;
+};
+
+const Wrapper = styled.div`
+  display: flex;
+`;
+
+const CopyIcon = styled.img.attrs<{ secondary?: boolean }>(
+  ({ theme: { icons }, secondary }) => ({
+    src: secondary ? icons.copySecondary : icons.copy,
+  })
+)<{ secondary?: boolean }>`
+  align-self: center;
+  width: 16px;
+  height: 15px;
+  margin: 0 3px;
+  cursor: pointer;
+  &:hover {
+    opacity: 0.5;
+  }
+  &:active {
+    transform: translate3d(2px, 2px, 0);
+  }
+`;
+
+const CopyButton = ({ onClickCopy, secondary, children, isCopied }: Props) => {
+  return (
+    <Wrapper onClick={onClickCopy}>
+      {children}
+      {!isCopied && <CopyIcon secondary={secondary} />}
```

Comment:
It seems to me that the proper name will be `isCustomRenderer` instead of `isCopied`, or in our case, we can just check that we have this custom renderer:
```suggestion
      {children}
      {!children && <CopyIcon secondary={secondary} />}
```

Reply:
Renamed. Exactly in this case where we have to show CopyIcon for 3-5 seconds depending on the timer. And after it shows CopyIcon again.  Not possible to reuse children.